cmd/deprecated-images.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"github.com/CircleCI-Public/circleci-cli/api"
+	"github.com/CircleCI-Public/circleci-cli/pipeline"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v3"
+)
+
+// CircleCI Linux VM images that will be permanately removed on May 31st.
+var deprecatedImages = []string{
+	"circleci/classic:201710-01",
+	"circleci/classic:201703-01",
+	"circleci/classic:201707-01",
+	"circleci/classic:201708-01",
+	"circleci/classic:201709-01",
+	"circleci/classic:201710-02",
+	"circleci/classic:201711-01",
+	"circleci/classic",
+	"circleci/classic:latest",
+	"circleci/classic:edge",
+	"circleci/classic:201808-01",
+	"ubuntu-1604:201903-01",
+	"ubuntu-1604:202004-01",
+	"ubuntu-1604:202007-01",
+	"ubuntu-1604:202010-01",
+	"ubuntu-1604:202101-01",
+	"ubuntu-1604:202104-01",
+}
+
+// Simplified Config -> job Structure for an image
+
+type job struct {
+	Machine interface{} `yaml:"machine"`
+}
+
+// Simplified Config Structure for an image
+type processedConfig struct {
+	Jobs map[string]job `yaml:"jobs"`
+}
+
+// Processes the config down to v2.0, then checks image used against the block list
+func deprecatedImageCheck(opts configOptions, flags *pflag.FlagSet, path string) error {
+
+	var params pipeline.Parameters
+
+	orgSlug, _ := flags.GetString("org-slug")
+
+	// first process the config so we can get all images
+	response, err := api.ConfigQuery(opts.cl, path, orgSlug, params, pipeline.LocalPipelineValues())
```

Comment:
Ah, good idea to process the config first, since Reusable Executors make this a lot more complicated to try to detect otherwise. :+1:

Reply:
👍🏼